src/pages/journal/__tests__/journal.spec.ts

```diff
@@ -145,5 +145,23 @@ describe('JournalPage', () => {
         expect(fixture.debugElement.query(By.css('#testReportPracticeMode'))).toBeNull();
       });
     });
+
+    describe('end to end practice mode', () => {
+      it('should show the end to end practice mode banner when config is set to true', () => {
+        component.showEndToEndPracticeMode =
+          jasmine.createSpy('showEndToEndPracticeMode').and.returnValue(true);
+
+        fixture.detectChanges();
+        expect(fixture.debugElement.query(By.css('#endToendPracticeMode'))).toBeDefined();
```

Comment:
`toBeDefined` is not a good test for existence when querying elements from the fixture, as `null` passes the check (i.e. no element found).

Reply:
Didn't know that will update